BSP/STM32F413/Src/BSP_ADC.c

```diff
@@ -0,0 +1,116 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+#include "BSP_ADC.h"
+#include "stm32f4xx.h"
+
+static volatile uint16_t ADCresults[2];
+
+static void ADC_InitDMA(void) {
+	// Start the clock for the DMA
+	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_DMA2, ENABLE);
+
+	// Create the DMA structure
+	DMA_InitTypeDef DMA_InitStruct;
+
+	DMA_InitStruct.DMA_Channel = DMA_Channel_0;
+	DMA_InitStruct.DMA_PeripheralBaseAddr = (uint32_t)&(ADC1->DR);
+	DMA_InitStruct.DMA_Memory0BaseAddr = (uint32_t) &ADCresults;
+	DMA_InitStruct.DMA_DIR = DMA_DIR_PeripheralToMemory;
+	DMA_InitStruct.DMA_BufferSize = 2;
+	DMA_InitStruct.DMA_PeripheralInc = DMA_PeripheralInc_Disable;
+	DMA_InitStruct.DMA_MemoryInc = DMA_MemoryInc_Enable;
+	DMA_InitStruct.DMA_PeripheralDataSize = DMA_PeripheralDataSize_HalfWord;
+	DMA_InitStruct.DMA_MemoryDataSize = DMA_MemoryDataSize_HalfWord;
+	DMA_InitStruct.DMA_Mode = DMA_Mode_Circular;
+	DMA_InitStruct.DMA_Priority = DMA_Priority_High;
+	DMA_InitStruct.DMA_FIFOMode = DMA_FIFOMode_Disable;
+	DMA_InitStruct.DMA_FIFOThreshold = DMA_FIFOThreshold_HalfFull;
+	DMA_InitStruct.DMA_MemoryBurst = DMA_MemoryBurst_Single;
+	DMA_InitStruct.DMA_PeripheralBurst = DMA_PeripheralBurst_Single;
+	DMA_Init(DMA2_Stream0, &DMA_InitStruct);
+
+	// Enable DMA2 stream 0
+	DMA_Cmd(DMA2_Stream0, ENABLE);
+}
+
+/**
+ * @brief   Initializes the ADC module. This is to measure the hall effect sensors
+ *          on the Current Monitor Board.
+ * @param   None
+ * @return  None
+ */
+void BSP_ADC_Init(ADC_t ADC_Channel) {
```

Comment:
I'm not sure I really thought about the initialization properly for the STM when I made this type. If it's not used to do anything different in the simulator, we don't need to pass this as a parameter to the Init function.